src/lib/storage/chunk.cpp

```diff
@@ -2,30 +2,64 @@
 
 #include "abstract_segment.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  _segments.push_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == _segments.size(), "There must be the same amount of values as in the segment!");
+  for (auto i = size_t{0}; i < values.size(); ++i) {
+    auto segment = _segments[i];
+    {
+      auto segment_pointer = dynamic_cast<ValueSegment<std::string>*>(segment.get());
+      if (segment_pointer != nullptr) {
+        segment_pointer->append(values[i]);
+      }
+    }
+    {
+      auto segment_pointer = dynamic_cast<ValueSegment<int32_t>*>(segment.get());
+      if (segment_pointer != nullptr) {
+        segment_pointer->append(values[i]);
+      }
+    }
+    {
+      auto segment_pointer = dynamic_cast<ValueSegment<int64_t>*>(segment.get());
+      if (segment_pointer != nullptr) {
+        segment_pointer->append(values[i]);
+      }
+    }
+    {
+      auto segment_pointer = dynamic_cast<ValueSegment<float>*>(segment.get());
+      if (segment_pointer != nullptr) {
+        segment_pointer->append(values[i]);
+      }
+    }
+    {
+      auto segment_pointer = dynamic_cast<ValueSegment<double>*>(segment.get());
+      if (segment_pointer != nullptr) {
+        segment_pointer->append(values[i]);
+      }
```

Comment:
Consider using a declaration in the if-clause:
```suggestion
      if (auto * const segment_pointer = dynamic_cast<ValueSegment<double*>(segment.get()) {
        segment_pointer->append(values[i]);
      }
```